src/lib/mongoBackend/mongoRegistrationsGet.cpp

```diff
@@ -0,0 +1,317 @@
+* If we have more than one, then the Registration is made in API V1 as this is not
+* possible in V2 and we cannot respond to the request using the current implementation of V2.
+* This function will be changed to work in a different way once issue #3044 is dealt with.
+*
+*/
+static bool setDataProvided(ngsiv2::Registration* regP, const mongo::BSONObj& r, bool arrayAllowed)
+{
+  std::vector<mongo::BSONElement> crV = getFieldF(r, REG_CONTEXT_REGISTRATION).Array();
+
+  if (crV.size() > 1)
+  {
+    return false;
+  }
+
+  //
+  // Extract the first (and only) CR from the contextRegistration vector
+  //
+  mongo::BSONObj cr0 = crV[0].embeddedObject();
+
+  setEntities(regP, cr0);
+  setAttributes(regP, cr0);
+  setProvider(regP, cr0);
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpires -
+*/
+static void setExpires(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->expires = (r.hasField(REG_EXPIRATION))? getIntFieldF(r, REG_EXPIRATION) : -1;
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*/
+static void setStatus(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->status = (r.hasField(REG_STATUS))? getStringFieldF(r, REG_STATUS): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setForwardingInformation -
+*/
+static void setForwardingInformation(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  // No forwarding info until API V2 forwarding is implemented
+}
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationsGet - 
+*/
+void mongoRegistrationsGet
+(
+  std::vector<ngsiv2::Registration>*  regV,
+  const std::string&                  tenant,
+  const std::vector<std::string>&     servicePathV,
+  OrionError*                         oeP
+)
+{
+  bool         reqSemTaken = false;
+  std::string  err;
+  mongo::OID   oid;
+  StatusCode   sc;
+
+  reqSemTake(__FUNCTION__, "Mongo Get Registrations", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo Get Registrations"));
+
+  std::auto_ptr<mongo::DBClientCursor>  cursor;
+  mongo::BSONObj                        q;
+
+  q = BSON(REG_SERVICE_PATH << fillQueryServicePath(servicePathV));
+
+  TIME_STAT_MONGO_READ_WAIT_START();
+  mongo::DBClientBase* connection = getMongoConnection();
+  if (!collectionQuery(connection, getRegistrationsCollectionName(tenant), q, &cursor, &err))
+  {
+    releaseMongoConnection(connection);
+    TIME_STAT_MONGO_READ_WAIT_STOP();
+    reqSemGive(__FUNCTION__, "Mongo Get Registrations", reqSemTaken);
+    oeP->fill(SccReceiverInternalError, err);
+    return;
+  }
+  TIME_STAT_MONGO_READ_WAIT_STOP();
+
+  /* Process query result */
+  while (moreSafe(cursor))
+  {
+    mongo::BSONObj        r;
+    ngsiv2::Registration  reg;
+
+    if (!nextSafeOrErrorF(cursor, &r, &err))
```

Comment:
Let's use the same strategy than subscriptions. In the case of error, not returning but loggin the error and continue, e.g:

```
    if (!nextSafeOrErrorF(cursor, &r, &err))
    {
      LM_E(("Runtime Error (exception in nextSafe(): %s - query: %s)", err.c_str(), q.toString().c_str()));
      continue;
    }
```

Reply:
Fixed in 72f4e9f